python_modules/dagster/dagster/_core/definitions/selector.py

```diff
@@ -313,3 +313,96 @@ def to_graphql_input(self):
             "repositoryName": self.repository_name,
             "partitionSetName": self.partition_set_name,
         }
+
+
+@whitelist_for_serdes
+class PartitionRangeSelector(
+    NamedTuple(
+        "_PartitionRangeSelector",
+        [("start", str), ("end", str)],
+    )
+):
+    """The information needed to resolve a partition range."""
+
+    def __new__(cls, start: str, end: str):
+        return super(PartitionRangeSelector, cls).__new__(
+            cls,
+            start=check.inst_param(start, "start", str),
+            end=check.inst_param(end, "end", str),
+        )
+
+    def to_graphql_input(self):
+        return {
+            "start": self.start,
+            "end": self.end,
+        }
+
+    @staticmethod
+    def from_graphql_input(graphql_data):
+        return PartitionRangeSelector(
+            start=graphql_data["start"],
+            end=graphql_data["end"],
+        )
+
+
+class PartitionsSelector(
+    NamedTuple(
+        "_PartitionsSelector",
+        [("partition_range", PartitionRangeSelector)],
+    )
+):
+    """The information needed to define selection partitions.
+    Using partition_range as property name to avoid shadowing Python 'range' builtin .
+    """
+
+    def __new__(cls, partition_range: PartitionRangeSelector):
+        return super(PartitionsSelector, cls).__new__(
+            cls,
+            partition_range=check.inst_param(partition_range, "range", PartitionRangeSelector),
+        )
+
+    def to_graphql_input(self):
+        return {
+            "range": self.range.to_graphql_input(),
+        }
+
+    @staticmethod
+    def from_graphql_input(graphql_data):
+        return PartitionsSelector(
+            partition_range=PartitionRangeSelector.from_graphql_input(graphql_data["range"])
+        )
+
+
+@whitelist_for_serdes
+class PartitionsByAssetSelector(
+    NamedTuple(
+        "PartitionsByAssetSelector",
+        [
+            ("asset_key", AssetKey),
+            ("partitions", Optional[PartitionsSelector]),
```

Comment:
Similar to the graphQL side, I think we don't need the `PartitionsSelector` object.

Can we just pass a `PartitionKeyRange` in instead? So we don't have to create a `PartitionRangeSelector` object as well

Reply:
Do these objects have to have the @whitelist_for_serdes?

Reply:
@clairelin135  Honestly, I wasn't sure whether we need @whitelist_for_serdes or not. I added it with the assumption that it is needed for serialization/deserialization in case we need to do that. However, if that is not necessary, I can remove them.

Reply:
I removed the decorator in my latest commits